client/my-sites/plugins/plugins-list/plugins-list-dataviews.tsx

```diff
@@ -0,0 +1,76 @@
+import { DataViews, filterSortAndPaginate } from '@wordpress/dataviews';
+import { useTranslate } from 'i18n-calypso';
+import React, { useEffect, useMemo, useState } from 'react';
+import { initialDataViewsState } from 'calypso/a8c-for-agencies/components/items-dashboard/constants';
+import { DataViewsState } from 'calypso/a8c-for-agencies/components/items-dashboard/items-dataviews/interfaces';
+import { useSelector } from 'calypso/state';
+import { Plugin } from 'calypso/state/plugins/installed/types';
+import { getPluginActionStatuses } from '../plugin-management-v2/utils/get-plugin-action-statuses';
+import { useActions } from './use-actions';
+import { useFields } from './use-fields';
+
+import './style.scss';
+
+interface Props {
+	currentPlugins: Array< Plugin >;
+	initialSearch?: string;
+	isLoading: boolean;
+	onSearch?: ( search: string ) => void;
+	bulkActionDialog: ( action: string, plugins: Array< Plugin > ) => void;
+}
+
+export default function PluginsListDataViews( {
+	currentPlugins,
+	initialSearch,
+	isLoading,
+	onSearch,
+	bulkActionDialog,
+}: Props ) {
+	const allStatuses = useSelector( getPluginActionStatuses );
+	const translate = useTranslate();
+
+	const fields = useFields( bulkActionDialog );
+	const actions = useActions( bulkActionDialog );
+
+	const [ dataViewsState, setDataViewsState ] = useState< DataViewsState >( () => ( {
+		...initialDataViewsState,
+		perPage: 15,
+		search: initialSearch,
+		fields: [ 'plugins', 'sites', 'update' ],
+	} ) );
+
+	// When search changes, notify the parent component
+	useEffect( () => {
+		onSearch && onSearch( dataViewsState.search || '' );
```

Comment:
So, this is how we trigger a data recalculation when the dataviews state changes. I've noticed [the fields define some filters](https://github.com/Automattic/wp-calypso/blob/04e61486e434796d0da8518c874ec3949d4f3c9d/client/my-sites/plugins/plugins-list/use-fields.tsx#L23), don't they affect the data (currentPlugins)?

Reply:
Jumped into a quick call with @paulopmt1 to understand this better. This is for serializing the user search into a URL parameter, so we can have this experience:

https://github.com/user-attachments/assets/72742681-ba4b-4f76-999b-3508726a3694

DataViews's search would only target the `plugins` field (name) as per the [field config](https://github.com/Automattic/wp-calypso/blame/d948ceb02c4468eecaf5d6765c2246c4b161db4c/client/my-sites/plugins/plugins-list/use-fields.tsx#L48), so all is good.

Reply:
For the record, this callback would bubble up to [PluginsList's doSearch prop](https://github.com/Automattic/wp-calypso/blob/d948ceb02c4468eecaf5d6765c2246c4b161db4c/client/my-sites/plugins/main.jsx#L314) that is [added](https://github.com/Automattic/wp-calypso/blob/d948ceb02c4468eecaf5d6765c2246c4b161db4c/client/my-sites/plugins/main.jsx#L449) by the [UrlSearch component](https://github.com/Automattic/wp-calypso/blob/ab2dc12fbf992095d48e4190097ed8fe4fbf8884/client/lib/url-search/index.js#L51) that handles it.